### DISCUSSION TITLE
Interaction with non-ebitenui interactive elements

### DISCUSSION THREAD

**Discussion by rom1mouret** [ebitenui/ebitenui#158] [Q&A]
Hi,

What's the idiomatic way of adding Ebitenui widgets on top of an existing interactive Ebitenui base?

For example, the base is a pannable world map with clickable countries. If Ebitenui opens a window, the countries drawn below the window should not be clickable, and the map should not be pannable by clicking within the boundaries of the window.

thanks

**Comment by mcarpenter622** [comment]
So a couple things
1) New Window widgets are on a different input layer than other objects in the UI so if you click inside a window, the base container objects should not be getting cursor events.
2) If you have multiple window widgets, it should work as expected, where each new window is on a higher "layer" than the previous ones. If not see 
```
// This option sets order the window will be drawn
//   - &lt; 0 will have the window drawn before the container
//   - &gt;= 0 will have the window drawn after the container
func (o WindowOptions) DrawLayer(layer int) WindowOpt {
	return func(w *Window) {
		w.DrawLayer = layer
	}
}
```
2) If you are trying to see if ebitenui is currently hovered over (because you're handling interaction with non ebitenui elements separately, there is a new PR that gives a new field you can check input.UIActive. (see https://github.com/ebitenui/ebitenui/pull/156)

**Reply by rom1mouret** [reply]
Thanks! So, if I understand correctly, the current idiomatic way is to put the pannable map and clickable countries inside an Ebitenui's container on the lowest layer of the UI?

> If you are trying to see if ebitenui is currently hovered over (because you're handling interaction with non ebitenui elements separately, there is a new PR that gives a new field you can check input.UIActive.

That's cool. I think that suits my case better. Can `input.UIActive` be used in the main `Update` (after `g.ui.Update()` ) or a copy of the boolean has to be stored in [game](https://github.com/ebitenui/ebitenui/blob/27496c28f409386546c7c4b6765642880ee35d75/_examples/demo/main.go) for this logic to work?
